news/2016-10-01-tip1-logging.asciidoc

```diff
@@ -0,0 +1,102 @@
+Weld does not log that much information with this level.
+You will have to set the log level to `DEBUG` (or even `TRACE`) to see what's going on in your application.
+If you set the level globally (for all frameworks and libraries), the log files will be bloated and it will be very hard to get some relevant info from there.
+If you set the level for Weld only (i.e. for `org.jboss.weld`), it should be better.
+Moreover, it's possible to use categories to filter messages from a particular "domain".
+Weld log messages are grouped into several categories:
+
+* `org.jboss.weld.Bootstrap`
+* `org.jboss.weld.Validator`
+* `org.jboss.weld.Bean`
+* `org.jboss.weld.Context`
+* `org.jboss.weld.El`
+* `org.jboss.weld.Conversation`
+* and others (see also link:https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/logging/Category.java[org.jboss.weld.logging.Category, window="_blank"])
+
+So for example, if you're only interested in log messages related to *EL resolution* set the log level for the `org.jboss.weld.El` category to `TRACE` (or level with corresponding priority).
+
+[[bootstrap]]
+== Monitoring bootstrap
+
+CDI is designed to fail fast.
+An extensive validation is performed during application initialization in order to avoid malicious errors at runtime.
+Container usually also log a lot of interesting info during bootstrap.
+Let's try to set the log level to *DEBUG* for the `org.jboss.weld.Bootstrap` category and see what we get.
+
+=== Registered beans
+
+As simple as it looks - Weld logs _all the beans_ found in the application, e.g.:
+
+----
+DEBUG [Bootstrap] WELD-000106: Bean: Managed Bean [class org.jboss.cdi.tck.tests.lookup.dependency.resolution.broken.unsatisfied.Vanilla] with qualifiers [@Any @Default]
+DEBUG [Bootstrap] WELD-000106: Bean: Producer Method [String[]] with qualifiers [@BatchProperty @Any] declared as [[UnbackedAnnotatedMethod] @Produces @BatchProperty public org.jberet.creation.BatchBeanProducer.getStringArray(InjectionPoint)]
+----
+=== Actions performed by extensions
+
+There is no doubt that portable extensions are really powerful.
+They can change almost anything.
+That's great for framework/libraries developers.
+On the other hand, this may cause problems when looking for a bug.
+There is an application class provided by a developer but what does the CDI container actually see?
+Well, it depends... because portable extensions are allowed to redefine this.
+And that's why Weld logs all the _"modification"_ operations performed by extensions, such as `ProcessAnnotatedType.veto()`:
+
+----
+DEBUG [Bootstrap] WELD-000148: ProcessAnnotatedType.veto() called by com.foo.MyExtension@50fed5b1 for [BackedAnnotatedType] public class com.foo.Foo
+----
+
+TIP: If your application deploys successfully, you can use link:http://docs.jboss.org/weld/reference/latest/en-US/html/devmode.html#devtools[Development Tools] to inspect the runtime info (more to be discussed in one of the next articles).
```

Comment:
`(more to be discussed in one of the next articles)` -> `(will be covered in one of the follow-up articles)`